### ISSUE TITLE
Support emitting preprocessed .svelte files

### ISSUE THREAD

**Issue by dominikg** [sveltejs/vite-plugin-svelte#475]
### Describe the problem

svelte-package currently does not read vite config at all and doesn't understand our `useVitePreprocess` option as a result.

This can lead to a divergence in library projects that use vite plugins like unocss or windicss for styles which look just fine in the preview during `pnpm dev`, but preprocessed output isn't working and a companion global css file can't be created efficiently

See https://github.com/unocss/unocss/pull/1725 for a draft of a custom preprocessor to solve it for unocss, but the problem is of a more general nature and we might want to offer tooling so that it can be solved with less effort.

#### 1. idea by @bluwy: a new `?preprocess` directive
```
const server = createServer()
const output = await server.ssrLoadModule('/User/foo/Component.svelte?preprocess')
console.log(output.preprocessed)
```
Which allows svelte-package and friends to glob .svelte files and then call ssrLoadModule on them to get the preprocessed output of each one

#### 2. a custom build option
add an option to output preprocessed .svelte files to a separate directory alongside compiled code via a rollup output hook
The output of `vite build` can then be used to get the preprocessed code

### Alternatives considered

Somehow add a preprocessor to svelte.config.js that allows uses vite+svelte under the hood, kind of the inverse of the above, but this risks inception

### Importance

nice to have

**Comment by dominikg** [comment]
cc @bluwy  @dummdidumm @ignatiusmb

**Comment by jacob-8** [comment]
Thank you, @dominikg! Since Vite plugins are becoming so valuable and useful as of late, it would be great if they could be used for `dev`, `build`, **and** `svelte-package`.

Let me add a little context. Since https://github.com/unocss/unocss/pull/1725 is mostly duplicated code (fork of a fork) waiting on https://github.com/unocss/unocss/pull/1692 to be merged, here's the [svelte preprocessor](https://github.com/unocss/unocss/pull/1725/files#diff-2659a5b7fafcce63280b3bc2d8ef1dd63bc58a57016b3a270b040bc23058427bR32-R81) I had to write which inits UnoCSS and then imports the transformSvelteSFC method. It actually wasn't too difficult to write this wrapper, but in the long run I'd rather use the Vite plugin instead for three main reasons:
- no need to support two methods of using UnoCSS's svelte-scoped mode (a Vite plugin for normal projects, and a Svelte preprocessor for component libraries that will be packaged)
- can take advantage of the Uno Inspector found at `/__unocss` when running the dev server. A Svelte preprocessor can't do this like a Vite plugin can.
- UnoCSS's custom transform directives can all be used. Due to how they work in Uno, I was not able to support them in the preprocessor wrapper and manually added just the one I needed (handle's `@apply` in style blocks)

I don't have enough knowledge to give advice on how to accomplish the task proposed in this issue but I can help test things out as helpful.

**Comment by dominikg** [comment]
related work: https://github.com/sveltejs/vite-plugin-svelte/tree/preprocess-api  contains an export for standalone preprocessors unsing vite config, these could be used to generate the preprocessed output without additional features in v-p-s.

It may still be nicer to offer a way to do this as part of a vite build to be able to produce extra resources alongside, eg the aforementioned styles generated by unocss.

**Comment by dominikg** [comment]
this is possible through the use or `?raw` query on import, see https://github.com/sveltejs/vite-plugin-svelte/blob/main/docs/advanced-usage.md#rawsvelte

given noone has asked used for this i'm not planning on pursuing this on a broader scale until there is actual demand. Feel free to request reopening if you have such a case
